### ISSUE TITLE
grpc configs (port, enabled etc) was not picked from the cloud config

### ISSUE THREAD

**Issue by ndhilliprasad** [LogNet/grpc-spring-boot-starter#99]
grpc-spring-boot-starter has made it easy to use grpc with spring boot. kudo's to all the contributors.

Lately I have been observing an issue after upgrading to the latest version of grpc-spring-boot-starter where the grpc.port (or the other configurations) was not pulled from the remote spring cloud config server. I looked at the spring-could-config and spring-boot git hub project issues before creating one here. It seems like GRpcServerEnvironment is not loading the configs from the remote config server. I found a interesting note from spring-cloud and spring-boot https://github.com/spring-cloud/spring-cloud-config/issues/1149 saying that EnvironmentPostProcessor might not work properly with cloud-config. Could some one please suggest a better approach to handle this issue

**Comment by jvmlet** [comment]
HI @ndhilliprasad , what is the actual behavior you are observing ? The  real port is random ? You  get  the startup error ? Please post your `application.yml` and  the actual behavior.

**Comment by ndhilliprasad** [comment]
Thanks for looking it to it @jvmlet. There is no error as such, app will just start with a random port. I have a bootstrap.yml which is configured to load the configurations from spring config server

```
spring:
  application:
    name: grpc-v1
  main:
    web-application-type: none
  profiles:
    active: local

  cloud:
    config:
      enabled: true
      failFast: true
      uri: http://xxxxx # config server which has all the application configurations

grpc:
  enableReflection: true
```
As a work around I have disabled the grpc (by adding following config to the bootstrap.yml) and exposing the bean for GRpcServerRunner in the configuration as follows. This is picking up the port from my config server

```
grpc:
  enabled: false
```

```java
    @Value("${grpc.port:7822}")
    private int port;

    @Bean
    @Primary
    public GRpcServerRunner grpcServerRunner(GRpcServerBuilderConfigurer configurer) {
        return new GRpcServerRunner(configurer, ServerBuilder.forPort(port));
    }
```

**Comment by gzukula** [comment]
We can't use versions above 2.4.0 because of this bug, since our gprc.port is configured via cloud config.

**Comment by jvmlet** [comment]
@ndhilliprasad  , @gzukula , I've just released `3.2.1-SNAPSHOT` that contains the fix + integration test with config server. Would you please  validate this version  at your  end before I release the `RELEASE` version ?

**Comment by jvmlet** [comment]
Fixed in  `3.2.1`
